tests/sentry/event_manager/test_event_manager_grouping.py

```diff
@@ -137,60 +136,48 @@ def test_updates_group_metadata(self):
-
-            with assume_test_silo_mode_of(AuditLogEntry):
-                audit_log_entry = AuditLogEntry.objects.get()
-
-            assert audit_log_entry.event == audit_log.get_event_id("PROJECT_EDIT")
-            assert audit_log_entry.actor_label == "Sentry"
-
-            assert audit_log_entry.data == {
-                "sentry:grouping_config": DEFAULT_GROUPING_CONFIG,
-                "sentry:secondary_grouping_config": LEGACY_CONFIG,
-                "sentry:secondary_grouping_expiry": ANY,  # tested separately below
-                "id": self.project.id,
-                "slug": self.project.slug,
-                "name": self.project.name,
-                "status": 0,
-                "public": False,
-            }
-
-            # When the config upgrade is actually happening, the expiry value is set before the
-            # audit log entry is created, which means the expiry is based on a timestamp
-            # ever-so-slightly before the audit log entry's timestamp, making a one-second tolerance
-            # necessary.
-            actual_expiry = audit_log_entry.data["sentry:secondary_grouping_expiry"]
-            expected_expiry = (
-                int(audit_log_entry.datetime.timestamp()) + SENTRY_GROUPING_UPDATE_MIGRATION_PHASE
-            )
-            assert actual_expiry == expected_expiry or actual_expiry == expected_expiry - 1
+        save_new_event({"message": "Adopt don't shop"}, self.project)
+        assert self.project.get_option("sentry:grouping_config") == DEFAULT_GROUPING_CONFIG
+
+        with assume_test_silo_mode_of(AuditLogEntry):
+            audit_log_entry = AuditLogEntry.objects.get()
+
+        assert audit_log_entry.event == audit_log.get_event_id("PROJECT_EDIT")
+        assert audit_log_entry.actor_label == "Sentry"
+
+        assert audit_log_entry.data == {
+            "sentry:grouping_config": DEFAULT_GROUPING_CONFIG,
+            "sentry:secondary_grouping_config": LEGACY_CONFIG,
+            "sentry:secondary_grouping_expiry": ANY,  # tested separately below
+            "id": self.project.id,
+            "slug": self.project.slug,
+            "name": self.project.name,
+            "status": 0,
+            "public": False,
+        }
+
+        # When the config upgrade is actually happening, the expiry value is set before the
+        # audit log entry is created, which means the expiry is based on a timestamp
+        # ever-so-slightly before the audit log entry's timestamp, making a one-second tolerance
+        # necessary.
+        actual_expiry = audit_log_entry.data["sentry:secondary_grouping_expiry"]
+        expected_expiry = (
+            int(audit_log_entry.datetime.timestamp()) + SENTRY_GROUPING_UPDATE_MIGRATION_PHASE
+        )
+        assert actual_expiry == expected_expiry or actual_expiry == expected_expiry - 1
 
-    def test_disabled_auto_update_does_not_update(self):
+    def test_does_not_update_project_in_exclude_list(self):
```

Comment:
Adding a new test for the exclude list. Once we have switched to the optimized code path we can remove it.